warehouse/query-core/src/test/java/datawave/query/index/lookup/IndexMatchTest.java

```diff
@@ -216,4 +223,46 @@ public void testHashCode() {
         IndexMatch indexMatch = new IndexMatch(uid);
         assertEquals(uid.hashCode(), indexMatch.hashCode());
     }
+    
+    @Test
+    public void testWriteReadIndexMatch() throws IOException {
+        IndexMatch match = new IndexMatch("uid");
+        
+        IndexMatch other = writeRead(match);
+        assertEquals("uid", other.uid);
+    }
+    
+    @Test(expected = NullPointerException.class)
+    public void testWriteReadIndexMatchWithNullUid() throws IOException {
+        IndexMatch match = new IndexMatch(null);
+        
+        // Cannot write a null uid.
+        IndexMatch other = writeRead(match);
+        assertNull(other.uid);
```

Comment:
This `assertNull` is probably never reached if the `match.write()` NPE's. I'd just drop it.

Reply:
Makes sense. I'll replace it with a fail("Should not have made it this far") kind of assertion.